### ISSUE TITLE
How do i judge the quatily of the model Tacotron2,such as accurary rate and recall,or other standard,Can you help me?

### ISSUE THREAD

**Issue by HenrySZP** [TensorSpeech/TensorFlowTTS#513]
I want to change based on this model，so i need know the judging criteria，pls help me.........

**Comment by machineko** [comment]
Most of the time u should just use [MAE](https://www.tensorflow.org/api_docs/python/tf/keras/losses/MAE)/[MSE](https://www.tensorflow.org/api_docs/python/tf/keras/losses/MSE) loss, but smaller loss from certain point dose not convert to audio quality (or mel reconstruction in case of Taco2). [ofc. in papers u'll get some sort of MOS metrics based on human reception of audio generated by models]
There are some ideas about better loss functions for end2end audio reconstruction for example [CDPAM](https://pixl.cs.princeton.edu/pubs/Manocha_2021_CCL/index.php) or older [DPAM](https://pixl.cs.princeton.edu/pubs/Manocha_2020_ADP/) but they are not implemented in this repo and u still need to test it :)

Most of they time i just listen to audio generated by waveglow + your mel generator (in ur case taco2)

**Comment by HenrySZP** [comment]
thank you，i'll try CDPAM ，how about mel_loss?if i use it , yes or no? @machineko

**Comment by machineko** [comment]
> mel_loss

Mel loss in the case of this repo is MAE/MSE  so u can use it without problems just remember that audio quality isn't comparable with the loss.

**Comment by machineko** [comment]
No problem!
You can close this issue if I respond to ur questions :)

**Comment by Koushik0901** [comment]
> Most of the time u should just use [MAE](https://www.tensorflow.org/api_docs/python/tf/keras/losses/MAE)/[MSE](https://www.tensorflow.org/api_docs/python/tf/keras/losses/MSE) loss, but smaller loss from certain point dose not convert to audio quality (or mel reconstruction in case of Taco2). [ofc. in papers u'll get some sort of MOS metrics based on human reception of audio generated by models]
> There are some ideas about better loss functions for end2end audio reconstruction for example [CDPAM](https://pixl.cs.princeton.edu/pubs/Manocha_2021_CCL/index.php) or older [DPAM](https://pixl.cs.princeton.edu/pubs/Manocha_2020_ADP/) but they are not implemented in this repo and u still need to test it :)
> 
> Most of they time i just listen to audio generated by waveglow + your mel generator (in ur case taco2)

@machineko can you let me know how you managed to implement mel generator to work with waveglow model? did you train the waveglow model or just used the pretrained waveglow?

**Comment by machineko** [comment]
> > Most of the time u should just use [MAE](https://www.tensorflow.org/api_docs/python/tf/keras/losses/MAE)/[MSE](https://www.tensorflow.org/api_docs/python/tf/keras/losses/MSE) loss, but smaller loss from certain point dose not convert to audio quality (or mel reconstruction in case of Taco2). [ofc. in papers u'll get some sort of MOS metrics based on human reception of audio generated by models]
> 
> > There are some ideas about better loss functions for end2end audio reconstruction for example [CDPAM](https://pixl.cs.princeton.edu/pubs/Manocha_2021_CCL/index.php) or older [DPAM](https://pixl.cs.princeton.edu/pubs/Manocha_2020_ADP/) but they are not implemented in this repo and u still need to test it :)
> 
> > 
> 
> > Most of they time i just listen to audio generated by waveglow + your mel generator (in ur case taco2)
> 
> 
> 
> @machineko can you let me know how you managed to implement mel generator to work with waveglow model? did you train the waveglow model or just used the pretrained waveglow?

I used preprocessing compatible with waveglow ( written in Jax )

**Comment by Koushik0901** [comment]
> > > Most of the time u should just use [MAE](https://www.tensorflow.org/api_docs/python/tf/keras/losses/MAE)/[MSE](https://www.tensorflow.org/api_docs/python/tf/keras/losses/MSE) loss, but smaller loss from certain point dose not convert to audio quality (or mel reconstruction in case of Taco2). [ofc. in papers u'll get some sort of MOS metrics based on human reception of audio generated by models]
> > 
> > 
> > > There are some ideas about better loss functions for end2end audio reconstruction for example [CDPAM](https://pixl.cs.princeton.edu/pubs/Manocha_2021_CCL/index.php) or older [DPAM](https://pixl.cs.princeton.edu/pubs/Manocha_2020_ADP/) but they are not implemented in this repo and u still need to test it :)
> > 
> > 
> > > 
> > 
> > 
> > > Most of they time i just listen to audio generated by waveglow + your mel generator (in ur case taco2)
> > 
> > 
> > @machineko can you let me know how you managed to implement mel generator to work with waveglow model? did you train the waveglow model or just used the pretrained waveglow?
> 
> I used preprocessing compatible with waveglow ( written in Jax )

can you point me to that preprocessing please?

**Comment by machineko** [comment]
> > > > Most of the time u should just use [MAE](https://www.tensorflow.org/api_docs/python/tf/keras/losses/MAE)/[MSE](https://www.tensorflow.org/api_docs/python/tf/keras/losses/MSE) loss, but smaller loss from certain point dose not convert to audio quality (or mel reconstruction in case of Taco2). [ofc. in papers u'll get some sort of MOS metrics based on human reception of audio generated by models]
> > > 
> > > 
> > > > There are some ideas about better loss functions for end2end audio reconstruction for example [CDPAM](https://pixl.cs.princeton.edu/pubs/Manocha_2021_CCL/index.php) or older [DPAM](https://pixl.cs.princeton.edu/pubs/Manocha_2020_ADP/) but they are not implemented in this repo and u still need to test it :)
> > > 
> > > 
> > > > 
> > > 
> > > 
> > > > Most of they time i just listen to audio generated by waveglow + your mel generator (in ur case taco2)
> > > 
> > > 
> > > @machineko can you let me know how you managed to implement mel generator to work with waveglow model? did you train the waveglow model or just used the pretrained waveglow?
> > 
> > 
> > I used preprocessing compatible with waveglow ( written in Jax )
> 
> can you point me to that preprocessing and the waveglow model please?

https://github.com/machineko/jaxaudio/blob/d5a8c8e55f090c26a8965d9463e424c0a8534b3c/jax_audio/preprocessing/transforms.py#L83

**Comment by Koushik0901** [comment]
thank you! @machineko
